src/tvheadend/Subscription.h

```diff
@@ -71,14 +72,19 @@ namespace tvheadend
     uint32_t          GetWeight() const;
     int32_t           GetSpeed() const;
     eSubsriptionState GetState() const;
+    std::string       GetProfile() const;
 
     /**
      * Subscribe to a channel on the backend
      * @param channelId the channel to subscribe to
      * @param weight the desired subscription weight
+     * @param profile the profile to use for the subscription
      * @param restart restart the current subscription (i.e. after lost connection), other parameters will be ignored
      */
-    void SendSubscribe(uint32_t channelId = 0, uint32_t weight = SUBSCRIPTION_WEIGHT_NORMAL, bool restart = false);
+    void SendSubscribe(uint32_t channelId = 0,
+                       uint32_t weight = SUBSCRIPTION_WEIGHT_NORMAL,
+                       const std::string &profile = "",
+                       bool restart = false);
 
```

Comment:
Is it possible to change the profile on run-time?
Otherwise we could just pass the profile with the constructor, that will make all this profile passing unnecessary?